### DISCUSSION TITLE
文档兼容性描述有关问题

### DISCUSSION THREAD

**Discussion by yoyo837** [hellof2e/quark-design#277] [General]
`slot`, `custom elements`, `shadow DOM` 这几个全部支持的最低版本, Chrome 应该是 54, iOS 应该是10.3吧? 另外, 安卓 7 里面的webview 是 52. 所以文档描述是否不太准确呢?

<img width="1199" alt="image" src="https://github.com/hellof2e/quark-design/assets/6134547/6894cc42-17ff-456f-8dab-d24ab799f043">

**Comment by yoyo837** [comment]
Edge “classic” 所有版本都不支持, 应该改为 `Edge "Chromium"`: `all` 对吗?

**Comment by sanqi-med** [comment]
> `slot`, `custom elements`, `shadow DOM` 这几个全部支持的最低版本, Chrome 应该是 54, iOS 应该是10.3吧? 另外, 安卓 7 里面的webview 是 52. 所以文档描述是否不太准确呢?
> 
> <img alt="image" width="1199" src="https://user-images.githubusercontent.com/6134547/250482875-6894cc42-17ff-456f-8dab-d24ab799f043.png">

嗯，chrome 54 有部分能力不支持，完全支持是从chrome 67哈。iOS的 safari 是 10.3 没啥问题，这个我修正下，上面的 safari 指的是mac 上的。

**Reply by yoyo837** [reply]
那再加个iOS Safari?

**Reply by yoyo837** [reply]
mac 上 10.1

**Comment by sanqi-med** [comment]
> Edge “classic” 所有版本都不支持, 应该改为 `Edge "Chromium"`: `all` 对吗?

Edge "Chromium 应该从 79 版本开始支持。
![WechatIMG623](https://github.com/hellof2e/quark-design/assets/18179784/b10cadc5-c637-4246-b87f-4ca75532dba5)

**Comment by sanqi-med** [comment]
> 那再加个iOS Safari?

嗯，下面那个就是移动端，上面的是桌面端哈，文档已修正
![WechatIMG624](https://github.com/hellof2e/quark-design/assets/18179784/934d4a7d-7449-4c2d-b5d9-9930f263f4d4)

**Comment by yoyo837** [comment]
安卓 7.1 里面的webview 是 52. 能支持吗?

**Comment by sanqi-med** [comment]
> 安卓 7.1 里面的webview 是 52. 能支持吗?
安卓每个厂家的webview 不完全是chrome内核。基本上 安卓 7以上都没什么问题，我们集团有统计工具，部分非主流机子安卓 7会有问题，不过现在都什么年代了，这种机子几乎没有了，web components 的兼容问题现在几乎可以忽略。

**Reply by yoyo837** [reply]
我最近运气爆棚就遇到了,  一款客户选择的大厅自助设备供应商, 就是安卓7的. 现有方案对接不进去. 无奈.

**Reply by yoyo837** [reply]
想套一下腾讯X5, 但遇到别的阻碍.

**Comment by sanqi-med** [comment]
> 我最近运气爆棚就遇到了, 一款客户选择的大厅自助设备供应商, 就是安卓7的. 现有方案对接不进去. 无奈.

好吧，那这个有点难顶

**Comment by yanhaijing** [comment]
我看 caniuse，安卓4.4和IOS10 兼容性是有问题的，有人实验过吗，哈罗内部小伙伴又遇到兼容性问题吗？

**Reply by yoyo837** [reply]
我被安卓 7 都卡死了, 你还来个4.4, 别想了.

**Reply by yoyo837** [reply]
本来就不兼容你发的版本, 不存在兼容问题, 因为压根不支持这么低.

**Reply by yanhaijing** [reply]
我们还要兼容安卓4.4 ಥ_ಥ

**Reply by yanhaijing** [reply]
那我得考虑换方案了ಥ_ಥ

**Reply by yoyo837** [reply]
那么低别用这个. 会让你下不了班.

**Reply by yoyo837** [reply]
项目开始前不先看看方案行得通不吗? 你是开发了才发现用不了? 那你可惨了.

**Reply by yanhaijing** [reply]
想问一下，这个方案和svelte和solidjs比有啥优势呢，感觉svelte和solid编译成webcomponents也行，而且更成熟些，solid的方案和这个比更接近react语法

**Reply by yanhaijing** [reply]
还没搞了，正在调研，但是我发现所有webcomponents方案都pass了，只能搞不成熟的mitosis了

**Reply by yoyo837** [reply]
velte和solid我没去了解, 它们编译成的webcomponents运行时也不依赖其他东西的话, 那这三者就区别不大.

**Reply by yanhaijing** [reply]
这两个都是没有虚拟dom的，所以是没有运行时依赖的

**Reply by yanhaijing** [reply]
stencli方案这种还是有很小的虚拟dom的，quarkc我是在掘金大会了解到的，感觉也不错，jsx，语法上看起来不太hook哈哈

**Reply by yoyo837** [reply]
有没有虚拟dom, 和有没有运行时依赖没有关系吧?

**Reply by yoyo837** [reply]
也许你和我有类似的业务需要, 互相认识一下, 交流沟通?

**Reply by yanhaijing** [reply]
有虚拟dom，运行时依赖肯定大，没有不一定小，需要在编译时做更工作，运行时才会小，比较适合这种webcomponents的场景

**Reply by yanhaijing** [reply]
加个微信 yanhaijing123
